ansible_base/authentication/authenticator_plugins/azuread.py

```diff
@@ -37,10 +37,25 @@ class AzureADConfiguration(BaseAuthenticatorConfiguration):
         ui_field_label=_('OIDC Secret'),
     )
 
+    GROUPS_CLAIM = CharField(
+        help_text=_("The JSON key used to extract the user's groups from the ID token or userinfo endpoint."),
+        required=False,
+        allow_null=False,
+        default="Group",
+        ui_field_label=_("Groups Claim"),
+    )
+
 
 class AuthenticatorPlugin(SocialAuthMixin, SocialAuthValidateCallbackMixin, AzureADOAuth2, AbstractAuthenticatorPlugin):
     configuration_class = AzureADConfiguration
     type = "azuread"
     logger = logger
     category = "sso"
     configuration_encrypted_fields = ['SECRET']
+
+    @property
+    def groups_claim(self):
+        return self.setting('GROUPS_CLAIM')
```

Comment:
```suggestion
        return getattr(self.setting, 'GROUPS_CLAIM', 'Groups')
```

Trying to think if there is a posibility of "I have an AD Authenticator already" so its never been saved with the new configuration schema so it might not have GROUP_CLAIM? Is this something you can test to see if we need to maybe change this to return a default if not found?

Reply:
The `GROUPS_CLAIM` field does not allow blanks so existing users can not have the authenticator configured without it. They will either have a "Group" value or `null` in there (but they have to explicitly do this via API to get `null` in there as there is no way to set it to `null` via UI). 
The `null` will become `None` in the python code and we already check for that in the pipeline function:
```
groups_claim = backend.groups_claim if backend.groups_claim is not None else "Group"
```

Reply:
I was thinking more of an exiting authenticator. 
i.e. Before your patch I have AzureAD configured and the "configuration" for that will be stored in the database (with no knowledge of GROUPS_CLAIM).
Then I apply your patch and someone tries to login with the configurator. IDK how `self.setting('GROUPS_CLAIM')` will behave in this scenario.

Reply:
I have tested that scenario even though it is an edge case because users can only do it via API.

First, tried creating the authenticator without claim specified:
Since the field "GROUPS_CLAIM does not allow blanks, the default value is used and the field is set to `Group`.

Then I relaxed the field constraints to allow `null` because that's what existing deployments would have and added a logging call to see what's returned from that getter. I've changed the `GROUPS_CLAIM` to `null` in the configuration of the authenticator.
Logging in with that authenticator worked fine, the getter returned None and our code in `social_auth.py` checks for that and uses `Group` in that case.

Reply:
I realized my test above was **NOT** testing the same behavior because I was always creating the authenticator configuration with updated code.

Today I have retested this as follows:
1. Created the Azure AD authentication without my changes and ended up with configuration that does not include the "GROUPS_CLAIM" field
2. Added back my changes
3. Logged in with Azure AD and the code in `social_auth.py` checked for None in group claims and used the default of "Groups" since it was `None`. I confirmed this by checking the logs
4. Updated the the authentication to set the "Groups Claim" to "groups" and added mapping that will make the test user an admin/superuser when an expected group are present.
5. Logged in again and the test user became admin/superuser